### DISCUSSION TITLE
Correct way to upgrade a current cluster.

### DISCUSSION THREAD

**Discussion by i5Js** [k3s-io/k3s#2982] [Q&A]
Hello all, 

I have a pretty stable K3s HA cluster, and I was thinking to update it to the last version. I've seen in the official documentation, you can stop the service, download the new k3s binary and start the service again, but, I would like to know if there is another better way.

Thanks.

**Comment by brandond** [comment]
You can also re-run the install script, as long as you pass along the same arguments as you initially used. That just replaces the binary though. Simply replacing the K3s binary and restarting is how things like the system-upgrade-controller do it, and is how I upgrade my nodes.

Is there something in particular you're looking for?

**Comment by i5Js** [comment]
Hi @brandond many thanks, no I was not looking for any special, just asking which was the optimal. I was checking the system-upgrade-controller as well...
